### ISSUE TITLE
Boost Software License?

### ISSUE THREAD

**Issue by vinniefalco** [HowardHinnant/date#513]
Why isn't this using the Boost Software License?

**Comment by HowardHinnant** [comment]
It wasn't a choice on this chart: http://cl.ly/5nAo
;-)

**Comment by dateuser** [comment]
It's a choice on this chart: https://i.imgur.com/SpmmbHq.png
:-)

**Comment by beached** [comment]
Is this a possibility? I would like to relicense using BSL for my library depending on this but I think date being MIT means that any binaries of my code will need the notice anyways.  Also, awesome work.

**Comment by HowardHinnant** [comment]
The MIT license does not require distribution with binaries.

Anyone who wants can **add** the BSL license to this source code.  But the MIT license can not be removed.  There are too many authors and contributors to do that now.

**Comment by vinniefalco** [comment]
> The MIT license does not require distribution with binaries.

Definitely false. Failing to include a copy of the copyright notice in binaries is a license violation.

https://opensource.stackexchange.com/questions/4058/what-is-the-point-of-including-the-mit-copyright-text-if-you-use-someones-code/4061#4061

This is exactly why MIT is less permissive than BSL. There are companies where the Boost Software License is accepted, but the MIT license is not.

**Comment by HowardHinnant** [comment]
It appears there is disagreement on this issue:  http://llvm.org/docs/DeveloperPolicy.html#legacy-license-structure

> are also licensed under the MIT License, which does not contain the binary redistribution clause.

And not being a lawyer myself, I have little interest in debating it.

As the main copyright holder of `date`, I have no intention of requiring the license attribution on non-source code.  Additionally, it is my hope that `date` will become obsolete within the new few years since it will be supplied by your C++ vendor.

**Comment by vinniefalco** [comment]
> It appears there is disagreement on this issue

Yep, the problem is that it is not crystal-clear whether or not binaries require attribution. Thus, a corporation will usually err on the side of caution and assume that attribution is required. The BSL was specifically developed to eliminate the ambiguity.

> I have little interest in debating it.

Right, and debates are pointless anyway because only actual trials and judgements can provide legal clarity.

> ...it is my hope that date will become obsolete within the new few years

Yep! I agree. But I am on a mission to discourage the world from choosing the MIT License for new projects, and to switch away from the MIT License for existing projects where it is practical to do so (which might not be applicable to <date>).

**Comment by pdimov** [comment]
> As the main copyright holder of `date`, I have no intention of requiring the license attribution on non-source code.

Then you shouldn't have used a license that requires it.

**Comment by roalz** [comment]
FYI re-licensing is possible, but it requires agreement from all (major) contributors (i.e. may not include one-line changes or non substantial changes to documentation, etc...).
But it may take long time and maybe be stuck if a few contributors are not reachable anymore,
see:
https://github.com/zeromq/libzmq/issues/2376
it's going on since 2017-03-19...
